src/core/exec-invoke.c

```diff
@@ -4367,6 +4367,10 @@ static void prepare_terminal(
               p->stdout_fd >= 0))
                 return;
 
+        /* Let's explicitly determine whether to reset via ANSI sequences or not, taking our ExecContext
+         * information into account */
+        bool use_ansi = exec_context_shall_ansi_seq_reset(context);
```

Comment:
Why called here? Why not call when `terminal_reset_defensive()` is called, like the other case in the below?

Reply:
a later commit will explain that: we want to use the result of this call also to condition the OSC sequence we generate. hence it makes sense to store the result in a local variable we can use at both places.